maps/maps.go

```diff
@@ -132,6 +132,31 @@ func Merge(a, b map[string]interface{}) {
 	}
 }
 
+// Erase erases the value present at a given path, from the map. The path
+// is the key map slice, for eg:, parent.child.key -> [parent child key].
+// Any empty, nested map on the path, is recursively deleted.
+//
+// It's important to note that all nested maps should be
+// map[string]interface{} and not map[interface{}]interface{}.
+// Use IntfaceKeysToStrings() to convert if necessary.
+func Erase(mp map[string]interface{}, path []string) {
+	next, ok := mp[path[0]]
+	if ok {
+		if len(path) == 1 {
+			delete(mp, path[0])
+			return
+		}
+		switch next.(type) {
```

Comment:
I think we can avoid the type assertions inside the switch if we do something like: `n := next.(type)` and inside the case, use `Erase(n, path[1:])`